xgcm/grid.py

```diff
@@ -1145,6 +1145,33 @@ def derivative(self, da, axis, **kwargs):
         dx = self.get_metric(diff, ("X",))
         return diff / dx
 
+    @docstrings.dedent
+    def integrate(self, da, axis, **kwargs):
+        """
+        Integrate along specified axis or axes.
+        Grid metrics (e.g. cell length, area, volume) are determined automatically.
+
+        Parameters
+        ----------
+        axis : str, list of str
+            Name of the axis on which to act
+        %(neighbor_binary_func.parameters.no_f)s
+
+        Returns
+        -------
+        da_i : xarray.DataArray
+            The integrated data
+        """
+
+        weight = self.get_metric(da, axis)
+        weighted = da * weight
+        # TODO: We should integrate xarray.weighted once available.
+
+        dim = [
+            di for ax in axis for di in self.axes[ax].coords.values() if di in da.dims
+        ]
```

Comment:
What happens if I try to integrate over an axis that is not present in the array? Like `grid.integrate(da, ('X', 'Y, 'Z'))` where da is 2D?